lib/wlib/strings/DynamicString.h

```diff
@@ -0,0 +1,476 @@
+	 *
+	 * @param val char string to add
+	 * @return the current string
+	 */
+	DynamicString &operator+= (const char* val){
+		return append(val);
+	}
+
+	/**
+	 * Modifier operator adds @code DynamicString object to the current string.
+	 *
+	 * @param other @code DynamicString string to add
+	 * @return the current string
+	 */
+	DynamicString &operator+= (DynamicString &other){
+		return append(other.c_str());
+	}
+
+	/**
+	 * Appends a character to the current string.
+	 *
+	 * @param c character to add
+	 * @return the current string
+	 */
+	DynamicString &append(const char c){
+		char array[2] = {c, '\0'};
+		return append(array);
+	}
+
+	/**
+	 * Appends a character string to the current string.
+	 *
+	 * @param str character string to add
+	 * @return the current string
+	 */
+	DynamicString &append(const char *str){
+		uint16_t bufferLength = this->length();
+		uint16_t newLength = (uint16_t)(bufferLength + strlen(str));
+		m_buffer = (char *)memory_realloc(m_buffer, (newLength + 1) * sizeof(char));
+		
+		for (int i = bufferLength; i < newLength; i++) {
+			m_buffer[i] = str[i-bufferLength];
+		}
+		
+		m_buffer[newLength] = '\0';
+
+		return *this;
+	}
+
+	/**
+	 * Appends a DynamicString string to the current string.
+	 *
+	 * @param str DynamicString string to add
+	 * @return the current string
+	 */
+	DynamicString &append(const DynamicString &str){
+		return append(str.c_str());
+	}
+
+	/**
+	 * Appends a character to the current string.
+	 *
+	 * @param c character to add
+	 * @return the current string
+	 */
+	DynamicString &push_back(const char c){
+		char array[2] = {c, '\0'};
+		return append(array);
+	}
+
+	/**
+     * Deletes the element @p pos from the String
+     *
+     * @param pos position of the element to be deleted
+     * @return the modified String
+     */
+    DynamicString &erase(uint16_t pos = 0){
+		uint16_t len = this->length();
+        if (len == 0 || pos >= len) return *this;
+
+		len--;
+
+        for (uint16_t i = pos; i < len; ++i) {
+            m_buffer[i] = m_buffer[i+1];
+        }
+
+        m_buffer[len] = '\0';
+        return *this;
+    }
+
+    /**
+     * Deletes the last character in the String
+     */
+    void pop_back(){
+		uint16_t len = this->length();
+        if (len == 0) return;
+
+        m_buffer[len - 1] = '\0';
+    }
+
+	/**
+	 * Provides access to character array that string uses behind the screen
+	 *
+	 * @return character array
+	 */
+	const char* c_str() const{
+		return m_buffer;
+	}
+
+	/**
+	 * Makes substring of the current string
+	 *
+	 * @param pos starting position
+	 * @param length length of the new string
+	 * @return new string which is a substring of current string
+	 */
+	DynamicString substr(uint16_t pos, uint16_t length) const{
+		char newBuffer[length + 1];
+
+		for(uint16_t i = pos; i < pos + length ;i++){
+			newBuffer[i-pos] = m_buffer[i];
+		}
+
+		newBuffer[length] = '\0';
+
+		DynamicString s{newBuffer};

```

Comment:
This function allocates and copies the substring and then has to copy it again in order to return. But we can address this later with `move` semantics. (Or you can try making this function return an rvalue)